### PR TITLE
 gh #87 dshdmiIn interface updates

### PR DESCRIPTION
**Pull request by Sumip98**
Updated interface file-related changes to the L1 code in dsHdmiIn

### CODE REVIEW

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_dsHdmiIn.c [hunk 1]
```diff
@@ -649,8 +649,9 @@ void test_l1_dsHdmiIn_negative_dsHdmiInScaleVideo(void) {
...
  * |:--:|-----------|----------|--------------|-----|
  * |01|Initialize the HDMI input sub-system using dsHdmiInInit() | | dsERR_NONE | Should Pass |
- * |02|Call dsHdmiInSelectZoomMode() and loop through all valid dsVideoZoom_t |[Valid Zoom Mode]| dsERR_NONE | Zoom mode should be updated |
- * |03|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
+ * |02|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_0, bool: false, dsVideoPlane_PRIMARY, bool false| dsERR_NONE | Port 0 should be selected |
+ * |03|Call dsHdmiInSelectZoomMode() and loop through all valid dsVideoZoom_t |[Valid Zoom Mode]| dsERR_NONE | Zoom mode should be updated |
+ * |04|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
  * 
  */
 void test_l1_dsHdmiIn_positive_dsHdmiInSelectZoomMode(void) {
```
[the remark this answers is not in the document]
Comment (Sumip98) [reply src/test_l1_dsHdmiIn.c]: Done

File: src/test_l1_dsHdmiIn.c [hunk 2]
```diff
@@ -659,13 +660,16 @@ void test_l1_dsHdmiIn_positive_dsHdmiInSelectZoomMode(void) {
     // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
     UT_ASSERT_EQUAL(dsHdmiInInit(), dsERR_NONE);
 
-    // Step 2: Loop through all dsVideoZoom_t values and call dsHdmiInSelectZoomMode()
```
[the remark this answers is not in the document]
Comment (Sumip98) [reply src/test_l1_dsHdmiIn.c:662]: Done

File: src/test_l1_dsHdmiIn.c [hunk 3]
```diff
@@ -2306,7 +2312,7 @@ void test_l1_dsHdmiIn_positive_dsGetEdid2AllmSupport(void) {
  * @note Testing for the `dsERR_OPERATION_NOT_SUPPORTED` and `dsERR_OPERATION_FAILED` might be challenging since it requires a specific scenario where the attempted operation is not supported or fails due to other reasons.
  * 
  */
-void test_l1_dsHdmiIn_negative_dsGetEdid2AllmSupport(void) {
+void test_l1_dsHdmiIn_negative_dsGetEdid2AllmSupport_sink(void) {
     gTestID = 50;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     bool allmSupport;
```
Comment (Anbukannadhasan) [inline src/test_l1_dsHdmiIn.c:2318]: test_l1_dsHdmiIn_register - In the register function, the changes to the function names are not being reflected. Additionally, please check the compilation part as well.

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: src/test_l1_dsHdmiIn.c (hunk 3 above)
Comment (Sumip98) [reply src/test_l1_dsHdmiIn.c:2318]: Done

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Review by akhilbhas — COMMENTED** [review]
I would recommend to create new UT_test_suite_t object for sink, source and common tests separately and add corresponding tests in to them using different api calls 

This would make it easier to enable or disable sink/source tests according to the test profile from main.c

Refer : https://github.com/rdkcentral/rdk-halif-test-hdmi_cec/blob/develop/src/main.c  on registering l2 tests

This comment will be addressed along with test profile changes.

**Review by akhilbhas — APPROVED** [review]
Looks good

**Pull request merged**